services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/PushStatusCollector.java

```diff
@@ -0,0 +1,194 @@
+package com.linkedin.venice.pushmonitor;
+
+import com.linkedin.venice.meta.ReadWriteStoreRepository;
+import com.linkedin.venice.meta.Store;
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.pushstatushelper.PushStatusStoreReader;
+import com.linkedin.venice.utils.Pair;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class PushStatusCollector {
+  private static final Logger LOGGER = LogManager.getLogger(PushStatusCollector.class);
+
+  private final Consumer<String> pushCompletedHandler;
+  private final BiConsumer<String, String> pushErrorHandler;
+  private final Map<String, TopicPushStatus> daVinciStoreTopicToPartitionCountMap = new VeniceConcurrentHashMap<>();
+
+  private final PushStatusStoreReader pushStatusStoreReader;
+
+  private final ReadWriteStoreRepository storeRepository;
+
+  private final int daVinciPushStatusScanPeriodInSeconds;
+
+  private final ScheduledExecutorService offlinePushCheckScheduler = Executors.newScheduledThreadPool(1);
+
+  public PushStatusCollector(
+      ReadWriteStoreRepository storeRepository,
+      PushStatusStoreReader pushStatusStoreReader,
+      Consumer<String> pushCompletedHandler,
+      BiConsumer<String, String> pushErrorHandler,
+      int daVinciPushStatusScanPeriodInSeconds) {
+    this.storeRepository = storeRepository;
+    this.pushStatusStoreReader = pushStatusStoreReader;
+    this.pushCompletedHandler = pushCompletedHandler;
+    this.pushErrorHandler = pushErrorHandler;
+    this.daVinciPushStatusScanPeriodInSeconds = daVinciPushStatusScanPeriodInSeconds;
```

Comment:
I am wondering whether we should introduce a feature flag to disable DVC push status collection or not.
In case DaVinci push status collection is problematic, it will affect all the pushes.
I am not sure, and what is your opinion on this?

Reply:
Do you mean a controller level flag to enable/disable collection of DVC status?

Reply:
yeah.
Since now, the backend version swap will depend on the DaVinci status as well, and if DVC status collection is not working properly, the impact will be huge.

Reply:
I see, in order to achieve this - also above comment to make other stuff configurable, I am going to add more argument to AbstarctPushMonitor constructor arg list, hopefully it is not too long, and if that's the case we can consider refactor it in future PR.

Reply:
Actually, let me do it in this PR. Just passing in VeniceControllerConfig can solve a lot of trouble.